### PR TITLE
Adding internal traffic policy work

### PR DESCRIPTION
**Pull request by marosset**
Signed-off-by: Mark Rossetti <marosset@microsoft.com>

Implementing internal traffic policy (Added into in-tree kube-proxy via https://github.com/kubernetes/kubernetes/pull/114407)
And also fixing some commented out code

### CODE REVIEW

**Comment by marosset** [comment]
I got load balancer rules working in a CAPZ cluster with these changes.

File: backend/kernelspace/proxier_events.go [hunk 1]
```diff
@@ -172,24 +174,26 @@ func (proxier *Proxier) newEndpointInfo(baseInfo *BaseEndpointInfo, _ *ServicePo
...
 	info := &serviceInfo{BaseServiceInfo: baseInfo}
 	preserveDIP := service.Annotations["preserve-destination"] == "true"
-	//localTrafficDSR := service.ExternalTrafficPolicy == v1.ServiceExternalTrafficPolicyTypeLocal
-	//err := hcn.DSRSupported()
-	//if err != nil {
-	//    preserveDIP = false
-	//    localTrafficDSR := false
-	//}
-	localTrafficDSR := false
+	localTrafficDSR := service.ExternalTrafficToLocal      // TODO: verify this is the correct field in the KPNG localv1 API
```
Comment (marosset) [inline backend/kernelspace/proxier_events.go:177]: @jayunit100 or @knabben - Is what I am doing the correct assignments for `localTrafficDSR` and `internalTrafficLocal` ?
These wre previously checking against core.v1 types

File: backend/kernelspace/proxier.go [hunk 2]
```diff
@@ -555,10 +555,10 @@ func (proxier *Proxier) syncProxyRules() {
 	// merge stale services gathered from updateEndpointsMap
 	for _, svcPortName := range endpointUpdateResult.StaleServiceNames {
 		klog.InfoS("echo %v", svcPortName)
-		//if svcInfo, ok := proxier.serviceMap[svcPortName]; ok && svcInfo != nil && svcInfo.Protocol() == v1.ProtocolUDP {
```
Comment (knabben) [inline backend/kernelspace/proxier.go:558]: isn't this required for UDP stale services and cleanup?

File: backend/kernelspace/proxier_events.go (hunk 1 above)
Comment (knabben) [reply backend/kernelspace/proxier_events.go:177]: I think it's correct, the upstream model is different from the kpng one related to these bool services parameters

**Pull request merged**